packages/KCatalog/KCatalogItem.vue

```diff
@@ -1,6 +1,6 @@
 <template>
   <KCard
-    :data-testid="item ? `${item.title.replace(' ', '-')}-catalog-item` : 'catalog-item'"
+    :data-testid="item && item.title ? `${item.title.replace(' ', '-')}-catalog-item` : 'catalog-item'"
```

Comment:
question: if you're replacing spaces to use as the testid, should you actually replace anything that isn't alpha-numeric? 
```suggestion
    :data-testid="item && item.title ? `${item.title.replace(/[^0-9a-z]/gi, '-')}-catalog-item` : 'catalog-item'"
```